Mathlib/RingTheory/TwoSidedIdeal/Matrix.lean

```diff
@@ -0,0 +1,153 @@
+    ∀ i j, x i j ∈ I :=
+  Iff.rfl
+
+variable [Inhabited ι]
+
+/-- If `I` is an two-sided ideal of `Mₙ(A)`, then `{M₀₀| M ∈ I}` is a two-sided ideal of `A`. -/
+@[simps!]
+def fromMatrix (I : TwoSidedIdeal M[ι, A]) :
+    TwoSidedIdeal A :=
+  .mk'
+    ((fun (x : M[ι, A]) ↦ x default default) '' I)
+    ⟨0, I.zero_mem, rfl⟩
+    (by rintro _ _ ⟨x, hx, rfl⟩ ⟨y, hy, rfl⟩; exact ⟨x + y, I.add_mem hx hy, rfl⟩)
+    (by rintro _ ⟨x, hx, rfl⟩; exact ⟨-x, I.neg_mem hx, rfl⟩)
+    (by
+      classical
+      rintro x _ ⟨y, hy, rfl⟩
+      exact ⟨Matrix.diagonal (fun _ ↦ x) * y, I.mul_mem_left _ _ hy, by simp⟩)
+    (by
+      classical
+      rintro _ y ⟨x, hx, rfl⟩
+      exact ⟨x * Matrix.diagonal (fun _ ↦ y), I.mul_mem_right _ _ hx, by simp⟩)
+
+variable [DecidableEq ι]
+
+/--
+The two-sided-ideals of `A` corresponds bijectively to that of `Mₙ(A)`.
+Given an ideal `I ≤ A`, we send it to `Mₙ(I)`.
+Given an ideal `J ≤ Mₙ(A)`, we send it to `{x₀₀ | x ∈ J}`.
+-/
+@[simps]
+def equivRingConMatrix : TwoSidedIdeal A ≃ TwoSidedIdeal M[ι, A] where
+  toFun I := I.mapMatrix A ι
+  invFun J := J.fromMatrix A
+  right_inv J := SetLike.ext fun x ↦ by
+    simp only [mem_mapMatrix]
+    constructor
+    · intro h
+      choose y hy1 hy2 using h
+      rw [Matrix.matrix_eq_sum_stdBasisMatrix x]
+      refine J.finsetSum_mem _ _ fun i _ ↦ J.finsetSum_mem _ _ fun j _ ↦ ?_
+      suffices
+          stdBasisMatrix i j (x i j) =
+          stdBasisMatrix i default 1 * y i j * stdBasisMatrix default j 1 by
+        rw [this]
+        exact J.mul_mem_right _ _ (J.mul_mem_left _ _ <| hy1 _ _)
+      ext a b
+      by_cases hab : a = i ∧ b = j
+      · rcases hab with ⟨ha, hb⟩
+        subst ha hb
+        simp only [StdBasisMatrix.apply_same, StdBasisMatrix.mul_right_apply_same,
+          StdBasisMatrix.mul_left_apply_same, one_mul, mul_one]
+        specialize hy2 a b
+        simp only [sub_zero] at hy2
+        exact hy2.symm
+      · conv_lhs =>
+          dsimp [stdBasisMatrix]
+          rw [if_neg (by tauto)]
+        rw [not_and_or] at hab
+        rcases hab with ha | hb
+        · rw [mul_assoc, Matrix.StdBasisMatrix.mul_left_apply_of_ne (h := ha)]
+        · rw [Matrix.StdBasisMatrix.mul_right_apply_of_ne (hbj := hb)]
+    · intro hx i j
+      refine ⟨stdBasisMatrix default i 1 * x * stdBasisMatrix j default 1,
+        J.mul_mem_right _ _ (J.mul_mem_left _ _ hx), ?_⟩
+      simp only [StdBasisMatrix.mul_right_apply_same, StdBasisMatrix.mul_left_apply_same, one_mul,
+        mul_one, sub_zero]
+  left_inv I := SetLike.ext fun x ↦ by
+    simp only
+    constructor
+    · intro h
+      choose y hy1 hy2 using h
+      simp only [sub_zero] at hy2
+      exact hy2 ▸ hy1 _ _
+    · intro h
+      exact ⟨of fun _ _ => x, by simp [h], by simp⟩
+
+/--
+`equivRingConMatrix` preserves order of two-sided ideals since if `I ≤ J` then
+clearly `Mₙ(I)` is a subset of `Mₙ(J)`.
+-/
+def OrderEquivMatrix : TwoSidedIdeal A ≃o TwoSidedIdeal M[ι, A] where
```

Comment:
Is this not the same as [TwoSidedIdeal.orderIsoMatricesOver](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Matrix/Ideal.html#TwoSidedIdeal.orderIsoMatricesOver) added in #17750?